### DISCUSSION TITLE
Deprecating macOS legacy - Calling Mac users of Jamulus

### DISCUSSION THREAD

**Discussion by softins** [jamulussoftware/jamulus#1657] [Announcements]
Edit May 04, 2023 by @ann0see 

Calling Mac users of Jamulus:
Are any of you still using the macOS legacy build (relying on compatibility with macOS 10.14 (Mojave, from 2018) ~~10.10 (Yosemite, from 2014) or macOS 10.11 (El Capitan, from 2015)~~)?

Please comment below and let us know.

**Comment by chrisrimple** [comment]
Not me, I've moved up to Mojave.  There are many Mac users in the music world that have not moved beyond Mojave yet because Catalina and later require 64-bit compatibility for all apps, and some apps/VSTs have been slow to move there; some may never do so if they're no longer being maintained.  If you're retaining support for Sierra and later, [that's compatible with hardware that's 10+ years old](https://en.wikipedia.org/wiki/MacOS_Sierra#System_requirements), and it's hard to believe that anyone is running Jamulus on something older.

**Reply by TerriQ716** [reply]
Hi, I've just seen this post and want to add that I have a Macbook Pro running MacOS High Sierra 10.13.6. I've just discovered Jamulus and completed the setup and tested it with another user. All leds were green, audio and latency were fine. Playing distance - Dorset to Scotland. And this was my first time😊👍. Would like to hear from other Mac users. Cheers.

**Comment by stevie2sheds** [comment]
Not myself personally, but I was trying to get a 10.9.5 (Mavericks) system going yesterday for someone. It was a mid-2012 macBook Pro so I got it to High Sierra (10.13.6) and it now works like a charm with an old Focusrite Saffire LE and the legacy version. Jamulus reports just 10ms latency above ping. That great performance of the older FireWire gear is why I've commented. Some folk will maintain a computer museum simply because they won't get a better performance with the new stuff. I personally run a FireWire Digidesign mBox 2 Pro on Ubuntu Studio and get 8ms on top of ping. That's an interface from 2006 connected to a PC well over 10 years old and I can't see how I would improve on that without spending mega bucks. Support of the old hardware is ultimately an OS thing, but there is good reason why folk will want to keep those FireWire devices and are reluctant to update both hardware and OS.

However the short answer to the original question is no!

**Reply by ann0see** [reply]
Thanks!

The issue is that legacy support introduces overhead and GitHub might in future decide to remove the tools to build legacy Jamulus.

**Reply by stevie2sheds** [reply]
That's absolutely understandable and understood. My thought on that is that like old drivers being available on the web, then previous .dmgs, .exes, and .debs of Jamulus versions could be downloadable. Sure, at some point old clients won't work with the latest servers, but if people really need to run older clients because of OS or hardware then they could run an older private server as well. The earlier versions are there on SourceForge.

**Reply by ann0see** [reply]
Yes. Also on Github.

**Comment by ann0see** [comment]
We're currently in a tricky situation where the legacy build
* Blocks us from automatically compiling language files which means that a separate step in the release process is needed
* Will force us to downgrade the Xcode version and SDK in future which might introduce bugs

I'd like to hear if the community would be OK with us dropping Legacy macOS support in the near future.

**Comment by ann0see** [comment]
This also means dropping support for Mojave unfortunately. We might be able to upgrade to some higher Qt 5 version to still support Mojave - this could still lead to negative consequences.
